openapi3/internalize_refs.go

```diff
@@ -0,0 +1,389 @@
+package openapi3
+
+import (
+	"context"
+	"path/filepath"
+	"strconv"
+	"strings"
+)
+
+type RefNameResolver func(string) string
+
+// DefaultRefResolver is a default implementation of refNameResolver for the
+// InternalizeRefs function.
+//
+// If a reference points to an element inside a document, it returns the last
+// element in the reference using filepath.Base. Otherwise if the reference points
+// to a file, it returns the file name trimmed of all extensions.
+func DefaultRefNameResolver(ref string) string {
+	if ref == "" {
+		return ""
+	}
+	split := strings.SplitN(ref, "#", 2)
+	if len(split) == 2 {
+		return filepath.Base(split[1])
+	}
+	ref = split[0]
+	for ext := filepath.Ext(ref); len(ext) > 0; ext = filepath.Ext(ref) {
+		ref = strings.TrimSuffix(ref, ext)
+	}
+	return filepath.Base(ref)
+}
+
+func schemaNames(s Schemas) []string {
+	out := make([]string, 0, len(s))
+	for i := range s {
+		out = append(out, i)
+	}
+	return out
+}
+
+func parametersMapNames(s ParametersMap) []string {
+	out := make([]string, 0, len(s))
+	for i := range s {
+		out = append(out, i)
+	}
+	return out
+}
+
+func isExternalRef(ref string) bool {
+	return ref != "" && !strings.HasPrefix(ref, "#/components/")
+}
+
+func (doc *T) addSchemaToSpec(s *SchemaRef, refNameResolver RefNameResolver) string {
```

Comment:
```suggestion
func (doc *T) addSchemaToSpec(s *SchemaRef, refNameResolver RefNameResolver) {
```
No call to this function actually use the returned value so let's drop that.